lib/node_modules/@stdlib/blas/base/drot/test/test.drot.js

```diff
@@ -0,0 +1,325 @@
+		[ 0.64, -1.26, 0.54, 0.2, -0.6, 0.2, 0.8 ],
+		[ 0.5, -0.9, 0.3, 0.7, -0.6, 0.2, 0.8 ],
+		[ 0.040000000000000036, -0.9, 0.3, 0.7, -0.6, 0.2, 0.8 ],
+		[ 0.040000000000000036, -0.9, 0.18, 0.7, -0.6, 0.2, 0.8 ],
+		[ 0.040000000000000036, -0.9, 0.18, 0.7, -0.18, 0.2, 0.16000000000000014 ] // eslint-disable-line max-len
+
+	];
+	nvalues = [ 0, 1, 2, 4 ];
+	ixvalues = [ 1, 2, -2, -1 ];
+	iyvalues = [ 1, -2, 1, -2 ];
+
+	xvalues = [
+		[ 0.6, 0.10, -0.5, 0.8, 0.9, -0.3, -0.4 ]
+	];
+	yvalues = [
+		[ 0.5, -0.9, 0.3, 0.7, -0.6, 0.2, 0.8 ]
+	];
+
+	c = 0;
+	for ( i = 0; i < ixvalues.length; i++ ) {
+		for ( j = 0; j < nvalues.length; j++ ) {
+			x = new Float64Array( xvalues[ 0 ] );
+			y = new Float64Array( yvalues[ 0 ] );
+			ix = ixvalues[ i ];
+			iy = iyvalues[ i ];
+			out = drot( nvalues[ j ], x, ix, y, iy, 0.8, 0.6 );
+			ex = new Float64Array( xexpected[ c ] );
+			ey = new Float64Array( yexpected[ c ] );
+			c += 1;
+			for ( k = 0; k < ex.length; k++ ) {
+				if ( x[ k ] === ex[ k ] ) {
+					t.strictEqual( x[ k ], ex[ k ], 'returns expected value' );
+				} else {
+					delta = abs( x[ k ] - ex[ k ] );
+					tol = 1.5 * EPS * abs( ex[ k ] );
+					t.ok( delta <= tol, 'within tolerance. x: '+x[ k ]+'. expected: '+ex[ k ]+'. delta: '+delta+'. tol: '+tol+'.' );
+				}
+				if ( y[ k ] === ey[ k ] ) {
+					t.strictEqual( y[ k ], ey[ k ], 'returns expected value' );
+				} else {
+					delta = abs( y[ k ] - ey[ k ] );
+					tol = 1.5 * EPS * abs( ey[ k ] );
+					t.ok( delta <= tol, 'within tolerance. y: '+y[ k ]+'. expected: '+ey[ k ]+'. delta: '+delta+'. tol: '+tol+'.' );
+				}
+				if ( out[ k ] === y[ k ] ) {
+					t.strictEqual( out[ k ], y[ k ], 'returns expected value' );
+				} else {
+					delta = abs( out[ k ] - y[ k ] );
+					tol = 1.5 * EPS * abs( y[ k ] );
+					t.ok( delta <= tol, 'within tolerance. out: '+out[ k ]+'. expected: '+y[ k ]+'. delta: '+delta+'. tol: '+tol+'.' );
+				}
+			}
+		}
+	}
+	t.end();
+});
+
+tape( 'the function supports an `x` stride', function test( t ) {
+	var xe;
+	var ye;
+	var x;
+	var y;
+	var N;
+
+	x = new Float64Array([
+		1.0, // 0
+		2.0,
+		3.0, // 1
+		4.0,
+		5.0  // 2
+	]);
+	y = new Float64Array([
+		6.0, // 0
+		7.0, // 1
+		8.0, // 2
+		9.0,
+		10.0
+	]);
+	N = 2;
+
+	drot( N, x, 2, y, 1, 0.707, 0.707 );
+
+	xe = new Float64Array( [ 4.949, 2.0, 7.07, 4.0, 5.0 ] );
+	ye = new Float64Array( [ 3.535, 2.828, 8.0, 9.0, 10.0 ] );
+
+	t.deepEqual( x, xe, 'returns expected value' );
+	t.deepEqual( y, ye, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the function supports a `y` stride', function test( t ) {
+	var xe;
+	var ye;
+	var x;
+	var y;
+	var N;
+
+	x = new Float64Array([
+		1.0, // 0
+		2.0, // 1
+		3.0, // 2
+		4.0,
+		5.0
+	]);
+	y = new Float64Array([
+		6.0, // 0
+		7.0,
+		8.0, // 1
+		9.0,
+		10.0 // 2
+	]);
+	N = 2;
+
+	drot( N, x, 1, y, 2, 0.707, 0.707 );
+
+	xe = new Float64Array( [ 4.949, 7.069999999999999, 3.0, 4.0, 5.0 ] );
```

Comment:
We should be very careful about using exact values like this. I'd prefer to use approximate equality tests, similar to the above using `EPS`.